libdnf-cli/utils/utf8.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef LIBDNF_CLI_UTILS_UTF8
+#define LIBDNF_CLI_UTILS_UTF8
+
+
+#include <string>
+
+
+namespace libdnf_cli::utils::utf8 {
+
+
+/// return length of an utf-8 encoded string
+std::size_t length(const std::string & str);
+
+
+/// return printable width of an utf-8 encoded string (considers non-printable and wide characters)
+std::size_t width(const std::string & str);
+
+
+/// return an utf-8 sub-string that matches specified character count
+std::string substr_length(const std::string & str, std::size_t pos, std::size_t len);
+
+
+/// return an utf-8 sub-string that matches specified printable width
+std::string substr_width(const std::string & str, std::size_t pos, std::size_t wid);
```

Comment:
std::size_t count = npos
Sometimes an user wants substring from pos to the end.
So, better is to support "default" value for length of the substring. Look at std::substr() for inspiration.

Reply:
fixed, test modified to use npos